### PR TITLE
Implement SPV_INTEL_arithmetic_fence

### PR DESCRIPTION
**Pull request by KornevNikita**
This extension creates support for a __fence keyword to prevent reordering and
reassociating operations.

Spec: https://github.com/intel/llvm/pull/3914
LLVM-review: https://reviews.llvm.org/D99675

### CODE REVIEW

**Comment by MrSidims** [comment]
Please add a test (though it would require some manual LLVM IR writing since https://reviews.llvm.org/D100118 isn't yet merged) and a reference to https://reviews.llvm.org/D99675 .

**Comment by KornevNikita** [comment]
@MrSidims thanks, forgot to add.

**Comment by KornevNikita** [comment]
@MrSidims @mlychkov @Fznamznon please take a look

**Review by MrSidims — APPROVED** [review]

**Comment by MrSidims** [comment]
Please resolve clang-format issues

File: lib/SPIRV/libSPIRV/SPIRVInstruction.h [hunk 1]
```diff
@@ -3127,6 +3127,21 @@ _SPIRV_OP(VariableLengthArray, true, 4)
 _SPIRV_OP(SaveMemory, true, 3)
 _SPIRV_OP(RestoreMemory, false, 2)
 #undef _SPIRV_OP
+
+class SPIRVArithmeticFenceINTELInst : public SPIRVInstTemplateBase {
```
Comment (mlychkov) [inline lib/SPIRV/libSPIRV/SPIRVInstruction.h:3131]: May SPIRVUnaryInst class be used as parent for SPIRVArithmeticFenceINTELInst ?

**Review by mlychkov — APPROVED** [review]

**Comment by MrSidims** [comment]
@KornevNikita there is yet one clang-format issue to address.

**Pull request merged**